common/persistence/Pinot/pinotVisibilityStore.go

```diff
@@ -0,0 +1,1111 @@
+	DocID                = "DocID"
+	DomainID             = "DomainID"
+	WorkflowID           = "WorkflowID"
+	RunID                = "RunID"
+	WorkflowType         = "WorkflowType"
+	CloseStatus          = "CloseStatus"
+	HistoryLength        = "HistoryLength"
+	TaskList             = "TaskList"
+	IsCron               = "IsCron"
+	NumClusters          = "NumClusters"
+	ShardID              = "ShardID"
+	Attr                 = "Attr"
+	StartTime            = "StartTime"
+	CloseTime            = "CloseTime"
+	UpdateTime           = "UpdateTime"
+	ExecutionTime        = "ExecutionTime"
+	Encoding             = "Encoding"
+	LikeStatement        = "%s LIKE '%%%s%%'"
+	IsDeleted            = "IsDeleted"
+
+	// used to be micro second
+	oneMicroSecondInNano = int64(time.Microsecond / time.Nanosecond)
+)
+
+type (
+	pinotVisibilityStore struct {
+		pinotClient         pnt.GenericClient
+		producer            messaging.Producer
+		logger              log.Logger
+		config              *service.Config
+		pinotQueryValidator *pnt.VisibilityQueryValidator
+	}
+
+	visibilityMessage struct {
+		DocID         string `json:"DocID,omitempty"`
+		DomainID      string `json:"DomainID,omitempty"`
+		WorkflowID    string `json:"WorkflowID,omitempty"`
+		RunID         string `json:"RunID,omitempty"`
+		WorkflowType  string `json:"WorkflowType,omitempty"`
+		TaskList      string `json:"TaskList,omitempty"`
+		StartTime     int64  `json:"StartTime,omitempty"`
+		ExecutionTime int64  `json:"ExecutionTime,omitempty"`
+		TaskID        int64  `json:"TaskID,omitempty"`
+		IsCron        bool   `json:"IsCron,omitempty"`
+		NumClusters   int64  `json:"NumClusters,omitempty"`
+		UpdateTime    int64  `json:"UpdateTime,omitempty"` // update execution,
+		ShardID       int64  `json:"ShardID,omitempty"`
+		// specific to certain status
+		CloseTime     int64 `json:"CloseTime,omitempty"`     // close execution
+		CloseStatus   int64 `json:"CloseStatus"`             // close execution
+		HistoryLength int64 `json:"HistoryLength,omitempty"` // close execution
+	}
+)
+
+var _ p.VisibilityStore = (*pinotVisibilityStore)(nil)
+
+func NewPinotVisibilityStore(
+	pinotClient pnt.GenericClient,
+	config *service.Config,
+	producer messaging.Producer,
+	logger log.Logger,
+) p.VisibilityStore {
+	return &pinotVisibilityStore{
+		pinotClient:         pinotClient,
+		producer:            producer,
+		logger:              logger.WithTags(tag.ComponentPinotVisibilityManager),
+		config:              config,
+		pinotQueryValidator: pnt.NewPinotQueryValidator(config.ValidSearchAttributes()),
+	}
+}
+
+func (v *pinotVisibilityStore) Close() {
+	// TODO: need to double check what is close trace do. Does it close the client?
+}
+
+func (v *pinotVisibilityStore) GetName() string {
+	return pinotPersistenceName
+}
+
+func (v *pinotVisibilityStore) RecordWorkflowExecutionStarted(
+	ctx context.Context,
+	request *p.InternalRecordWorkflowExecutionStartedRequest,
+) error {
+	v.checkProducer()
```

Comment:
move this check at creation `NewPinotVisibilityStore` to avoid runtime panic. Use log.Fatal instead.

Reply:
Fixed